packages/agents/cartographer/poller/src/lib/operations/stableswap.ts

```diff
@@ -0,0 +1,89 @@
+import { createLoggingContext, SubgraphQueryByTimestampMetaParams, StableSwapExchange } from "@connext/nxtp-utils";
+
+import { getContext } from "../../shared";
+
+const getMaxStableSwapTimestamp = (exchanges: StableSwapExchange[]): number => {
+  return exchanges.length == 0 ? 0 : Math.max(...exchanges.map((exchange) => exchange?.timestamp ?? 0));
+};
+
+export const updateStableSwap = async () => {
+  const {
+    adapters: { subgraph, database },
+    logger,
+    domains,
+  } = getContext();
+  const { requestContext, methodContext } = createLoggingContext("updateStableSwap");
+
+  const subgraphQueryMetaParams: Map<string, SubgraphQueryByTimestampMetaParams> = new Map();
+  const lastestBlockNumbers: Map<string, number> = await subgraph.getLatestBlockNumber(domains);
+
+  await Promise.all(
+    domains.map(async (domain) => {
+      const pools = await subgraph.getStableSwapPools(domain);
+      logger.info("Retrieved stable swap pools", requestContext, methodContext, {
+        domain,
+        pools: pools,
+        count: pools.length,
+      });
+
+      if (pools.length) {
+        await database.saveStableSwapPool(pools);
```

Comment:
Is there no need to checkpoint ?

Reply:
Yes, no need. because we get 100 pools for each domain. I believe that max pool count of domain is less than 100.